src/hooks/query-hook.tsx

```
@@ -47,4 +47,45 @@ export default function useQuery<TResponse, TVariables>(
     reload: executeQuery,
     loading
   };
+}
+
+export function useDeferredQuery<TResponse, TVariables>(
+  gqlQuery: DocumentNode,
+  onCompleted: (result: ApolloQueryResult<TResponse>) => void) {
+  const { loading: authLoading, isAuthenticated } = useAuth();
+
+  const apolloClient = useApolloClient();
+  const [loading, updateLoading] = useState(false);
+  const [error, updateError] = useState(false);
+
+  const executeQuery = useCallback(async (variables?: TVariables): Promise<void> => {
+    if (!authLoading && isAuthenticated) {
```

In the existing `useQuery` hook, this check if used to wait until authentication has completed before actually executing teh query. Here the invocation will just fail silently - consider adding an 'else' block here that logs an error to the console.

---

Now that this hook also reports error states, I've just set the result to error. The user can always retry. Also added a todo for logging